services/agents/registry.go

```diff
@@ -799,6 +806,28 @@ func (r *Registry) CheckConnectionToService(ctx context.Context, q *reform.Queri
 			Dsn:     agent.DSN(service, 2*time.Second, "", nil),
 			Timeout: ptypes.DurationProto(3 * time.Second),
 		}
+	case models.ExternalServiceType:
+		dsn := agent.DSN(service, 2*time.Second, "", nil)
+		if !agent.PushMetrics {
```

Comment:
I think all this stuff should be in agent_model.go

Reply:
After discussions extracted to another method.